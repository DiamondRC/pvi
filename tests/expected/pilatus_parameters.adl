
file {
    name="pilatus_parameters.adl"
    version=030109
}
display {
    object {
        x=391
        y=58
        width=310
        height=287
    }
    clr=14
    bclr=4
    cmap=""
    gridSpacing=5
    gridOn=0
    snapToGrid=0
}
"color map" {
    ncolors=65
    colors {
        ffffff,
        ececec,
        dadada,
        c8c8c8,
        bbbbbb,
        aeaeae,
        9e9e9e,
        919191,
        858585,
        787878,
        696969,
        5a5a5a,
        464646,
        2d2d2d,
        000000,
        00d800,
        1ebb00,
        339900,
        2d7f00,
        216c00,
        fd0000,
        de1309,
        be190b,
        a01207,
        820400,
        5893ff,
        597ee1,
        4b6ec7,
        3a5eab,
        27548d,
        fbf34a,
        f9da3c,
        eeb62b,
        e19015,
        cd6100,
        ffb0ff,
        d67fe2,
        ae4ebc,
        8b1a96,
        610a75,
        a4aaff,
        8793e2,
        6a73c1,
        4d52a4,
        343386,
        c7bb6d,
        b79d5c,
        a47e3c,
        7d5627,
        58340f,
        99ffff,
        73dfff,
        4ea5f9,
        2a63e4,
        0a00b8,
        ebf1b5,
        d4db9d,
        bbc187,
        a6a462,
        8b8239,
        73ff6b,
        52da3b,
        3cb420,
        289315,
        1a7309,
    }
}
rectangle {
    object {
        x=0
        y=4
        width=310
        height=25
    }
    "basic attribute" {
        clr=2
    }
}
text {
    object {
        x=0
        y=5
        width=310
        height=25
    }
    "basic attribute" {
        clr=54
    }
    textix="pilatus features - $(P)$(R)"
    align="horiz. centered"
}
# (Rectangle)
    rectangle {
        object {
            x=5
            y=40
            width=300
            height=56
        }
        "basic attribute" {
            clr=14
            fill="outline"
        }
    }

    rectangle {
        object {
            x=10
            y=45
            width=290
            height=20
        }
        "basic attribute" {
            clr=2
        }
    }
    text {
        object {
            x=5
            y=45
            width=290
            height=20
        }
        "basic attribute" {
            clr=54
        }
        textix="Top"
        align="horiz. centered"
    }

# (Rectangle)
    rectangle {
        object {
            x=5
            y=111
            width=300
            height=171
        }
        "basic attribute" {
            clr=14
            fill="outline"
        }
    }

    rectangle {
        object {
            x=10
            y=116
            width=290
            height=20
        }
        "basic attribute" {
            clr=2
        }
    }
    text {
        object {
            x=5
            y=116
            width=290
            height=20
        }
        "basic attribute" {
            clr=54
        }
        textix="Ancillary Information"
        align="horiz. centered"
    }

# (Static Text)
    text {
        object {
            x=10
            y=70
            width=145.0
            height=23
        }
        "basic attribute" {
            clr=14
        }
        textix="Armed"
        align="horiz. right"
    }

byte {
    object {
        x=220.0
        y=70
        width=20
        height=20
    }
    monitor {
        chan="$(P)$(R)Armed_RBV"
        clr=60
        bclr=64
    }
    sbit=0
}


# (Static Text)
    text {
        object {
            x=10
            y=141
            width=145.0
            height=23
        }
        "basic attribute" {
            clr=14
        }
        textix="Threshold Energy"
        align="horiz. right"
    }

# (Textentry)
    "text entry" {
        object {
            x=160.0
            y=141
            width=67.5
            height=19
        }
        control {
            chan="$(P)$(R)ThresholdEnergy"
            clr=14
            bclr=51
        }
        limits {
        }
    }

# (Textupdate)
    "text update" {
        object {
            x=232.5
            y=141
            width=67.5
            height=19
        }
        monitor {
            chan="$(P)$(R)ThresholdEnergy_RBV"
            clr=54
            bclr=4
        }
        align="horiz. left"
        limits {
        }
    }

# (Static Text)
    text {
        object {
            x=10
            y=164
            width=145.0
            height=23
        }
        "basic attribute" {
            clr=14
        }
        textix="Min Flat Field"
        align="horiz. right"
    }

# (Textentry)
    "text entry" {
        object {
            x=160.0
            y=164
            width=67.5
            height=19
        }
        control {
            chan="$(P)$(R)MinFlatField"
            clr=14
            bclr=51
        }
        limits {
        }
    }

# (Textupdate)
    "text update" {
        object {
            x=232.5
            y=164
            width=67.5
            height=19
        }
        monitor {
            chan="$(P)$(R)MinFlatField_RBV"
            clr=54
            bclr=4
        }
        align="horiz. left"
        limits {
        }
    }

# (Static Text)
    text {
        object {
            x=10
            y=187
            width=145.0
            height=23
        }
        "basic attribute" {
            clr=14
        }
        textix="Gap Fill"
        align="horiz. right"
    }

# (Menu Button)
    menu {
        object {
            x=160.0
            y=187
            width=67.5
            height=19
        }
        control {
            chan="$(P)$(R)GapFill"
            clr=14
            bclr=51
        }
    }

# (Textupdate)
    "text update" {
        object {
            x=232.5
            y=187
            width=67.5
            height=19
        }
        monitor {
            chan="$(P)$(R)GapFill_RBV"
            clr=54
            bclr=4
        }
        align="horiz. left"
        limits {
        }
    }

# (Static Text)
    text {
        object {
            x=10
            y=210
            width=145.0
            height=23
        }
        "basic attribute" {
            clr=14
        }
        textix="TVX Version"
        align="horiz. right"
    }

# (Textupdate)
    "text update" {
        object {
            x=160.0
            y=210
            width=140.0
            height=19
        }
        monitor {
            chan="$(P)$(R)TVXVersion_RBV"
            clr=54
            bclr=4
        }
        align="horiz. left"
        limits {
        }
    }

# (Static Text)
    text {
        object {
            x=10
            y=233
            width=145.0
            height=23
        }
        "basic attribute" {
            clr=14
        }
        textix="Pixel Cut Off"
        align="horiz. right"
    }

# (Textupdate)
    "text update" {
        object {
            x=160.0
            y=233
            width=140.0
            height=19
        }
        monitor {
            chan="$(P)$(R)PixelCutOff_RBV"
            clr=54
            bclr=4
        }
        align="horiz. left"
        limits {
        }
    }

# (Static Text)
    text {
        object {
            x=10
            y=256
            width=145.0
            height=23
        }
        "basic attribute" {
            clr=14
        }
        textix="CBF Header String"
        align="horiz. right"
    }

# (Textentry)
    "text entry" {
        object {
            x=160.0
            y=256
            width=140.0
            height=19
        }
        control {
            chan="$(P)$(R)HeaderString"
            clr=14
            bclr=51
        }
        limits {
        }
    }
