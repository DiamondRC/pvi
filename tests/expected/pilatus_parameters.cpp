PilatusParameters::PilatusParameters(asynPortDriver *parent) {
    parent->createParam("ThresholdEnergy", asynParamFloat64, &ThresholdEnergy);
    parent->createParam("MinFlatField", asynParamInt32, &MinFlatField);
    parent->createParam("GapFill", asynParamInt32, &GapFill);
    parent->createParam("TVXVersion", asynParamOctet, &TVXVersion);
    parent->createParam("PixelCutOff", asynParamInt32, &PixelCutOff);
    parent->createParam("HeaderString", asynParamOctet, &HeaderString);
    parent->createParam("Armed", asynParamInt32, &Armed);
}
