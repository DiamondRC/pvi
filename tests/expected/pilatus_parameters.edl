4 0 1
beginScreenProperties
major 4
minor 0
release 1
x 300
y 50
w 255
h 250
font "arial-bold-r-12.0"
ctlFont "arial-bold-r-12.0"
btnFont "arial-bold-r-12.0"
fgColor index 14
bgColor index 3
textColor index 14
ctlFgColor1 index 16
ctlFgColor2 index 25
ctlBgColor1 index 10
ctlBgColor2 index 3
topShadowColor index 1
botShadowColor index 11
title "pilatus features - $(P)$(R)"
showGrid
snapToGrid
gridSize 5
endScreenProperties

# (Group)
object activeGroupClass
beginObjectProperties
major 4
minor 0
release 0
x 0
y 0
w 255
h 30

beginGroup

# (Rectangle)
object activeRectangleClass
beginObjectProperties
major 4
minor 0
release 0
x 0
y 0
w 255
h 30
lineColor index 3
fill
fillColor index 3
endObjectProperties

# (Lines)
object activeLineClass
beginObjectProperties
major 4
minor 0
release 1
x 0
y 2
w 255
h 24
lineColor index 11
fillColor index 0
numPoints 3
xPoints {
  0 0
  1 255
  2 255
}
yPoints {
  0 26
  1 26
  2 2
}
endObjectProperties

# (Static Text)
object activeXTextClass
beginObjectProperties
major 4
minor 1
release 0
x 0
y 2
w 255
h 24
font "arial-bold-r-16.0"
fontAlign "center"
fgColor index 14
bgColor index 48
value "pilatus features - $(P)$(R)"
endObjectProperties

# (Lines)
object activeLineClass
beginObjectProperties
major 4
minor 0
release 1
x 0
y 2
w 255
h 24
lineColor index 1
fillColor index 0
numPoints 3
xPoints {
  0 0
  1 0
  2 255
}
yPoints {
  0 26
  1 2
  2 2
}
endObjectProperties

endGroup

endObjectProperties

# (Rectangle)
object activeRectangleClass
beginObjectProperties
major 4
minor 0
release 0
x 5
y 50
w 245
h 150
lineColor index 14
fill
fillColor index 5
endObjectProperties

# (Static Text)
object activeXTextClass
beginObjectProperties
major 4
minor 1
release 0
x 5
y 40
w 150
h 14
font "arial-medium-r-12.0"
fontAlign "center"
fgColor index 14
bgColor index 8
value "  Ancilliary Information  "
autoSize
border
endObjectProperties

# (Static Text)
object activeXTextClass
beginObjectProperties
major 4
minor 1
release 0
x 10
y 55
w 110
h 20
font "arial-bold-r-10.0"
fgColor index 14
bgColor index 3
useDisplayBg
value "Threshold Energy"

endObjectProperties

# (Textentry)
object TextentryClass
beginObjectProperties
major 10
minor 0
release 0
x 120
y 55
w 60
h 17
controlPv "$(P)$(R)ThresholdEnergy"
fgColor index 25
fgAlarm
bgColor index 3
fill
font "arial-bold-r-12.0"
endObjectProperties

# (Textupdate)
object TextupdateClass
beginObjectProperties
major 10
minor 0
release 0
x 185
y 55
w 60
h 17
controlPv "$(P)$(R)ThresholdEnergy_RBV"
fgColor index 16
fgAlarm
bgColor index 10
fill
font "arial-bold-r-12.0"
fontAlign "center"
endObjectProperties

# (Static Text)
object activeXTextClass
beginObjectProperties
major 4
minor 1
release 0
x 10
y 75
w 110
h 20
font "arial-bold-r-10.0"
fgColor index 14
bgColor index 3
useDisplayBg
value "Min Flat Field"

endObjectProperties

# (Textentry)
object TextentryClass
beginObjectProperties
major 10
minor 0
release 0
x 120
y 75
w 60
h 17
controlPv "$(P)$(R)MinFlatField"
fgColor index 25
fgAlarm
bgColor index 3
fill
font "arial-bold-r-12.0"
endObjectProperties

# (Textupdate)
object TextupdateClass
beginObjectProperties
major 10
minor 0
release 0
x 185
y 75
w 60
h 17
controlPv "$(P)$(R)MinFlatField_RBV"
fgColor index 16
fgAlarm
bgColor index 10
fill
font "arial-bold-r-12.0"
fontAlign "center"
endObjectProperties

# (Static Text)
object activeXTextClass
beginObjectProperties
major 4
minor 1
release 0
x 10
y 95
w 110
h 20
font "arial-bold-r-10.0"
fgColor index 14
bgColor index 3
useDisplayBg
value "Gap Fill"

endObjectProperties

# (Menu Button)
object activeMenuButtonClass
beginObjectProperties
major 4
minor 0
release 0
x 120
y 95
w 125
h 17
fgColor index 25
bgColor index 3
inconsistentColor index 0
topShadowColor index 1
botShadowColor index 11
controlPv "$(P)$(R)GapFill"
indicatorPv "$(P)$(R)GapFill_RBV"
font "arial-bold-r-12.0"
endObjectProperties

# (Static Text)
object activeXTextClass
beginObjectProperties
major 4
minor 1
release 0
x 10
y 115
w 110
h 20
font "arial-bold-r-10.0"
fgColor index 14
bgColor index 3
useDisplayBg
value "TVX Version"

endObjectProperties

# (Textupdate)
object TextupdateClass
beginObjectProperties
major 10
minor 0
release 0
x 120
y 115
w 125
h 17
controlPv "$(P)$(R)TVXVersion_RBV"
fgColor index 16
fgAlarm
bgColor index 10
fill
font "arial-bold-r-12.0"
fontAlign "center"
endObjectProperties

# (Static Text)
object activeXTextClass
beginObjectProperties
major 4
minor 1
release 0
x 10
y 135
w 110
h 20
font "arial-bold-r-10.0"
fgColor index 14
bgColor index 3
useDisplayBg
value "Pixel Cut Off"

endObjectProperties

# (Textupdate)
object TextupdateClass
beginObjectProperties
major 10
minor 0
release 0
x 120
y 135
w 125
h 17
controlPv "$(P)$(R)PixelCutOff_RBV"
fgColor index 16
fgAlarm
bgColor index 10
fill
font "arial-bold-r-12.0"
fontAlign "center"
endObjectProperties

# (Static Text)
object activeXTextClass
beginObjectProperties
major 4
minor 1
release 0
x 10
y 155
w 110
h 20
font "arial-bold-r-10.0"
fgColor index 14
bgColor index 3
useDisplayBg
value "Header String"

endObjectProperties

# (Textentry)
object TextentryClass
beginObjectProperties
major 10
minor 0
release 0
x 120
y 155
w 125
h 17
controlPv "$(P)$(R)HeaderString"
fgColor index 25
fgAlarm
bgColor index 3
fill
font "arial-bold-r-12.0"
endObjectProperties

# (Static Text)
object activeXTextClass
beginObjectProperties
major 4
minor 1
release 0
x 10
y 175
w 110
h 20
font "arial-bold-r-10.0"
fgColor index 14
bgColor index 3
useDisplayBg
value "Armed"

endObjectProperties

# (Byte)
object ByteClass
beginObjectProperties
major 4
minor 0
release 0
x 170
y 175
w 17
h 17
controlPv "$(P)$(R)Armed_RBV"
lineColor index 14
onColor index 15
offColor index 19
lineWidth 2
numBits 1
endObjectProperties
# (Exit Button)
object activeExitButtonClass
beginObjectProperties
major 4
minor 1
release 0
x 155
y 220
w 95
h 25
fgColor index 46
bgColor index 3
topShadowColor index 1
botShadowColor index 11
label "EXIT"
font "arial-bold-r-14.0"
3d
endObjectProperties

