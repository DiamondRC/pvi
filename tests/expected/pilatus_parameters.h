#ifndef PILATUS_PARAMETERS_H
#define PILATUS_PARAMETERS_H

class PilatusParameters {
public:
    PilatusParameters(asynPortDriver *parent);
    /* Group: AncilliaryInformation */
    int ThresholdEnergy;  /* asynParamFloat64 Setting */
}

#endif //PILATUS_PARAMETERS_H
