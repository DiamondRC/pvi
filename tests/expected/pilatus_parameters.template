# Group: AncilliaryInformation

record(ai, "$(P)$(R)ThresholdEnergy_RBV") {
    field(SCAN, "I/O Intr")
    field(DESC, "Threshold energy in keV")
    field(INP,  "@asyn($(PORT),$(ADDR),$(TIMEOUT))ThresholdEnergy")
    field(DTYP, "asynFloat64")
    field(EGU,  "keV")
    field(PREC, "3")
}

record(ao, "$(P)$(R)ThresholdEnergy") {
    field(DESC, "Threshold energy in keV")
    field(DTYP, "asynFloat64")
    field(EGU,  "keV")
    field(PREC, "3")
    field(OUT,  "@asyn($(PORT),$(ADDR),$(TIMEOUT))ThresholdEnergy")
    field(PINI, "YES")
    field(VAL,  "10.0")
    info(autosaveFields, "VAL")
}

record(longin, "$(P)$(R)MinFlatField_RBV") {
    field(SCAN, "I/O Intr")
    field(DESC, "Minimum flat field value")
    field(INP,  "@asyn($(PORT),$(ADDR),$(TIMEOUT))MinFlatField")
    field(DTYP, "asynInt32")
    field(EGU,  "Counts")
}

record(longout, "$(P)$(R)MinFlatField") {
    field(DESC, "Minimum flat field value")
    field(DTYP, "asynInt32")
    field(EGU,  "Counts")
    field(OUT,  "@asyn($(PORT),$(ADDR),$(TIMEOUT))MinFlatField")
    field(PINI, "YES")
    field(VAL,  "100")
    info(autosaveFields, "VAL")
}

record(mbbi, "$(P)$(R)GapFill_RBV") {
    field(SCAN, "I/O Intr")
    field(DESC, "GapFill")
    field(INP,  "@asyn($(PORT),$(ADDR),$(TIMEOUT))GapFill")
    field(DTYP, "asynInt32")
    field(ONST, "0")
    field(ONVL, "0")
    field(TWST, "-1")
    field(TWVL, "-1")
    field(ZRST, "N.A.")
    field(ZRVL, "2")
}

record(mbbo, "$(P)$(R)GapFill") {
    field(DESC, "GapFill")
    field(DTYP, "asynInt32")
    field(ONST, "0")
    field(ONVL, "0")
    field(TWST, "-1")
    field(TWVL, "-1")
    field(ZRST, "N.A.")
    field(ZRVL, "2")
    field(OUT,  "@asyn($(PORT),$(ADDR),$(TIMEOUT))GapFill")
    field(PINI, "YES")
    field(VAL,  "0")
    info(autosaveFields, "VAL")
}

record(stringin, "$(P)$(R)TVXVersion_RBV") {
    field(SCAN, "I/O Intr")
    field(DESC, "TVXVersion_RBV")
    field(INP,  "@asyn($(PORT),$(ADDR),$(TIMEOUT))TVXVersion")
    field(DTYP, "asynOctetRead")
}

record(ai, "$(P)$(R)PixelCutOff_RBV") {
    field(SCAN, "I/O Intr")
    field(DESC, "Readback for tau pixel cut off")
    field(INP,  "@asyn($(PORT),$(ADDR),$(TIMEOUT))PixelCutOff")
    field(DTYP, "asynInt32")
    field(EGU,  "Counts")
}

record(waveform, "$(P)$(R)HeaderString") {
    field(DESC, "HeaderString")
    field(DTYP, "asynOctetWrite")
    field(FTVL, "CHAR")
    field(NELM, "68")
    field(INP,  "@asyn($(PORT),$(ADDR),$(TIMEOUT))HeaderString")
    field(PINI, "YES")
    field(VAL,  "0")
    info(autosaveFields, "VAL")
}

record(bi, "$(P)$(R)Armed_RBV") {
    field(SCAN, "I/O Intr")
    field(DESC, "Armed flag")
    field(INP,  "@asyn($(PORT),$(ADDR),$(TIMEOUT))Armed")
    field(DTYP, "asynInt32")
    field(ONAM, "Armed")
    field(ZNAM, "Unarmed")
}

