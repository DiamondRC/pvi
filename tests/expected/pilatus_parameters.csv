Parameter,PVs,Description
*Top*
Armed,$(P)$(R)Armed_RBV,"Armed flag

Indicates Pilatus is ready for external triggers
"
*AncillaryInformation*
ThresholdEnergy,"$(P)$(R)ThresholdEnergy
$(P)$(R)ThresholdEnergy_RBV","Threshold energy in keV

camserver uses this value to set the discriminators in each pixel.
It is typically set to the incident x-ray energy ($(P)$(R)Energy),
but sometimes other values may be preferable.
"
MinFlatField,"$(P)$(R)MinFlatField
$(P)$(R)MinFlatField_RBV",Minimum flat field value
GapFill,"$(P)$(R)GapFill
$(P)$(R)GapFill_RBV",GapFill
TVXVersion,$(P)$(R)TVXVersion_RBV,TVXVersion_RBV
PixelCutOff,$(P)$(R)PixelCutOff_RBV,"Readback for tau pixel cut off
"
HeaderString,$(P)$(R)HeaderString,Header from CBF file
